### ISSUE TITLE
[feature request] Entity Unique Identifier Missing

### ISSUE THREAD

**Issue by v1k70rk4** [storm1er/ha_integration_parcelsapp#4]
Hi,

I really like the integration, and it works well.

I would like to request that the parcelsapp sensors have a unique identifier in the following format: `sensor.parcelsapp_<name>`.

This is necessary so I can filter the entities with a markdown card. If there's another way to achieve this, I'm happy to consider it :)

Once again, thank you for your work, it really fills a gap, as Track17 is basic compared to this.

Viktor

**Comment by v1k70rk4** [comment]
I'm currently using this, but the word "parcel" has to remain in the name.

It can be quickly rewritten into other languages as it includes a translation option. I took the basics from the widely used Track17 markdown script and adapted it for ParcelsApp.

And below you can find a simple package removal tool.

```
{% set computer_text = '<ha-icon icon="mdi:laptop" style="color:yellow"></ha-icon>' %}

# **[ParcelsApp](https://parcelsapp.com/)**

{% set parcels = states.sensor | selectattr('entity_id', 'match', 'sensor.parcelsapp_tracking_*') | list %}
{% set parcels_by_status = parcels | groupby('attributes.status') %}

{% set status_translation = {
    'archive': 'Archived packages',
    'delivered': 'Delivered packages',
    'transit': 'Packages in transit',
    'arrived': 'Packages awaiting delivery',
    'pickup': 'Packages awaiting pickup',
    'pending': 'Pending'
} %}

{% for status_group in parcels_by_status %}
{% set status = status_group.grouper %}
{% set packages = status_group.list %}

## {{ status_translation.get(status, status) | capitalize }}

---

{% for package in packages %}
### <ha-icon icon="mdi:package-variant" style="color:yellow"></ha-icon> **{{ package.attributes.get('name', '') }}** [{{computer_text}}](https://parcelsapp.com/en/tracking/{{ package.attributes.get('tracking_id', '') }})

{%- if package.attributes.get('tracking_id') not in [None, 'undefined'] %}
> **Tracking number**: {{ package.attributes.tracking_id }}
{%- endif %}
{%- if package.attributes.get('message') not in [None, 'undefined'] %}
> **Status**: {{ package.attributes.message }}
{%- endif %}
{%- if package.attributes.get('carrier') not in [None, 'undefined'] %}
> **Carrier**: {{ package.attributes.carrier }}
{%- endif %}
{%- if package.attributes.get('location') not in [None, 'undefined'] %}
> **Location**: {{ package.attributes.location }}
{%- endif %}
{%- if package.attributes.get('origin') not in [None, 'undefined'] %}
> **Origin**: {{ package.attributes.origin }}
{%- endif %}
{%- if package.attributes.get('destination') not in [None, 'undefined'] %}
> **Destination**: {{ package.attributes.destination }}
{%- endif %}
{%- if package.attributes.get('days_in_transit') not in [None, 'undefined'] %}
> **Days in transit**: {{ package.attributes.days_in_transit }}
{%- endif %}
{%- if package.attributes.get('uuid') not in [None, 'undefined'] %}
> **UUID**: {{ package.attributes.uuid }}
{%- endif %}
{%- if package.attributes.get('uuid_timestamp') not in [None, 'undefined'] %}
> **UUID timestamp**: {{ package.attributes.uuid_timestamp }}
{%- endif %}
{%- if package.attributes.get('last_updated') not in [None, 'undefined'] %}
> **Last updated**: {{ relative_time(package.attributes.last_updated) }}
{%- endif %}

{% endfor %}

{% endfor %}
```

![image](https://github.com/user-attachments/assets/dc74b27a-7649-4c87-94f4-0f02c68704c3)

```
type: custom:auto-entities
card:
  type: entities
  title: Package Removal
  state_color: true
filter:
  include:
    - domain: sensor
      entity_id: sensor.parcelsapp_tracking_*
      options:
        type: custom:button-card
        name: |
          [[[ return entity.attributes.name ]]]
        icon: mdi:package-variant-closed
        tap_action:
          action: call-service
          service: parcelsapp.remove_package
          service_data:
            tracking_id: |
              [[[ return entity.attributes.tracking_id ]]]
          confirmation:
            text: Are you sure you want to delete this package?
        layout: icon_name
        styles:
          card:
            - padding: 10px
          grid:
            - grid-template-areas: i status
            - grid-template-columns: min-content auto
            - align-items: right
            - padding-top: 2px
            - font-size: 14px
          name:
            - grid-area: "n"
            - justify-self: start
          custom_fields:
            status:
              - grid-area: status
              - justify-self: end
              - padding-top: 2px
              - font-size: 10px
        custom_fields:
          status: |
            [[[ return entity.attributes.tracking_id ]]]
```
![image](https://github.com/user-attachments/assets/492a9513-2491-4967-bcea-351d069c617b)

**Comment by storm1er** [comment]
Good catch, working on it (will be sensor.parcelsapp_tracking_<tracking_id> a custom name if given)

**Comment by storm1er** [comment]
Fixed in 1.1.2, you may need to remove/readd your parcels

**Comment by v1k70rk4** [comment]
> Fixed in 1.1.2, you may need to remove/readd your parcels

Thank you very much :)  
It’s working fine.
